### PR TITLE
Gradle cleanup, interface definition cleanup

### PR DESCRIPTION
**Pull request by mkosmul**
* Using Gradle toolchain for easier compilation on various machines (auto-downloading needed JDK)
* Updated dependencies and version catalog made more readable
* Removed unneccessary public and abstract modifiers from interface methods

### CODE REVIEW

File: src/main/java/oripa/domain/paint/byvalue/ByValueContext.java [hunk 1]
```diff
@@ -4,16 +4,16 @@
  * Copyright (C) 2005-2009 Jun Mitani         http://mitani.cs.tsukuba.ac.jp/
 
     This program is free software: you can redistribute it and/or modify
-    it under the terms of the GNU General Public License as published by
```
Comment (Ooouch1) [inline src/main/java/oripa/domain/paint/byvalue/ByValueContext.java:7]: The "public" in comments should not be removed.
I know putting a license comment in every code is an old way and annoying, though.

File: src/main/java/oripa/gui/view/DialogView.java [hunk 2]
```diff
@@ -4,16 +4,16 @@
  * Copyright (C) 2005-2009 Jun Mitani         http://mitani.cs.tsukuba.ac.jp/
 
     This program is free software: you can redistribute it and/or modify
-    it under the terms of the GNU General Public License as published by
```
Comment (Ooouch1) [inline src/main/java/oripa/gui/view/DialogView.java:7]: The "public" in comments should not be removed.

File: src/main/java/oripa/gui/view/estimation/EstimationResultFrameView.java [hunk 3]
```diff
@@ -4,16 +4,16 @@
  * Copyright (C) 2005-2009 Jun Mitani         http://mitani.cs.tsukuba.ac.jp/
 
     This program is free software: you can redistribute it and/or modify
-    it under the terms of the GNU General Public License as published by
```
Comment (Ooouch1) [inline src/main/java/oripa/gui/view/estimation/EstimationResultFrameView.java:7]: The "public" in comments should not be removed.

File: src/main/java/oripa/gui/view/main/KeyProcessing.java [hunk 4]
```diff
@@ -4,25 +4,25 @@
  * Copyright (C) 2005-2009 Jun Mitani         http://mitani.cs.tsukuba.ac.jp/
 
     This program is free software: you can redistribute it and/or modify
-    it under the terms of the GNU General Public License as published by
```
Comment (Ooouch1) [inline src/main/java/oripa/gui/view/main/KeyProcessing.java:7]: The "public" in comments should not be removed.

File: src/main/java/oripa/gui/view/main/MainFrameDialogFactory.java [hunk 5]
```diff
@@ -4,16 +4,16 @@
  * Copyright (C) 2005-2009 Jun Mitani         http://mitani.cs.tsukuba.ac.jp/
 
     This program is free software: you can redistribute it and/or modify
-    it under the terms of the GNU General Public License as published by
```
Comment (Ooouch1) [inline src/main/java/oripa/gui/view/main/MainFrameDialogFactory.java:7]: The "public" in comments should not be removed.

File: src/main/java/oripa/gui/view/model/ModelViewFrameView.java [hunk 6]
```diff
@@ -4,16 +4,16 @@
  * Copyright (C) 2005-2009 Jun Mitani         http://mitani.cs.tsukuba.ac.jp/
 
     This program is free software: you can redistribute it and/or modify
-    it under the terms of the GNU General Public License as published by
```
Comment (Ooouch1) [inline src/main/java/oripa/gui/view/model/ModelViewFrameView.java:7]: The "public" in comments should not be removed.

File: src/main/java/oripa/gui/view/model/ObjectGraphicDrawer.java [hunk 7]
```diff
@@ -4,16 +4,16 @@
  * Copyright (C) 2005-2009 Jun Mitani         http://mitani.cs.tsukuba.ac.jp/
 
     This program is free software: you can redistribute it and/or modify
-    it under the terms of the GNU General Public License as published by
```
Comment (Ooouch1) [inline src/main/java/oripa/gui/view/model/ObjectGraphicDrawer.java:7]: The "public" in comments should not be removed.

File: src/main/java/oripa/util/history/UndoInfo.java [hunk 8]
```diff
@@ -4,16 +4,16 @@
  * Copyright (C) 2005-2009 Jun Mitani         http://mitani.cs.tsukuba.ac.jp/
 
     This program is free software: you can redistribute it and/or modify
-    it under the terms of the GNU General Public License as published by
```
Comment (Ooouch1) [inline src/main/java/oripa/util/history/UndoInfo.java:7]: The "public" in comments should not be removed.

**Review by Ooouch1 — CHANGES_REQUESTED** [review]
Good updates. Leave the "public" in comments remaining, or remove all license comments from the code.

File: src/main/java/oripa/domain/paint/byvalue/ByValueContext.java (hunk 1 above)
Comment (mkosmul) [reply src/main/java/oripa/domain/paint/byvalue/ByValueContext.java:7]: Sorry, my oversight. I'll fix this in a moment.

**Review by Ooouch1 — APPROVED** [review]
Cool. You can merge this PR if allowed. (I'm not sure the project setting allows you to merge...)

**Comment by mkosmul** [comment]
> Cool. You can merge this PR if allowed. (I'm not sure the project setting allows you to merge...)
Unfortunately, I don't have permission to merge.

**Pull request merged**